src/item.cpp

```diff
@@ -197,25 +197,15 @@ item::item(JsonObject &jo)
     deserialize(jo);
 }
 
-item::~item()
-{
-}
-
-void item::make( const std::string new_type, bool scrub )
+item& item::convert( const itype_id& new_type )
 {
     type = find_type( new_type );
-    contents.clear();
-
-    if (scrub) {
-        components.clear();
-        charges = -1;
-        bday = 0;
-        name = "";
-        curammo = NULL;
+    if( is_null() ) {
```

Comment:
> If there is no such type defined, it creates one and sets its id accordingly.

But for that item `is_null() == true`?

> In fact, I would consider this message (and branch) here redundant

Yes, good point actually I'll drop that and update the function documentation